src/index.js

```diff
@@ -52,185 +41,159 @@ export default function lodash({ types }) {
+        }
+
+        return result;
       }
-    });
-  });
 
-  const replaceElements = replaceNodeList('elements');
-  const replaceArguments = replaceNodeList('arguments');
+      parentPath = parentPath.parentPath;
+    }
+  }
 
   /*--------------------------------------------------------------------------*/
 
-  const visitor = {
+  function importer({ declarations, defaults, members }) {
+    // Phase 1: Import all modules.
 
-    Program(path, state) {
-      const { ids } = _.assign(mapping, config(state.opts));
+    _.each(members, ({ pkgStore, spec }) => {
+      const { node } = spec;
+      const name = node.local.name;
 
-      if (_.isEmpty(ids)) {
-        throw new Error('Cannot find module');
+      if (name === 'chain') {
+        throw path.buildCodeFrameError(CHAIN_ERROR);
       }
-      _.each(ids, id => {
-        store.set(id);
-        mapping.modules.get(id).forEach(function(value, key) {
-          store.set(id + '/' + key);
+
+      const identifier = importModule(pkgStore, node.imported.name, spec.hub.file);
+      pkgStore.set(name, identifier);
+      pkgStore.set(identifier.name, identifier);
+    })
+    _.each(defaults, ({ pkgStore, spec }) => {
+        const binding = spec.scope.getBinding(spec.node.local.name);
+        _.each(binding.referencePaths, path => {
+          if (path.key === 'callee') {
+            throw path.buildCodeFrameError(CHAIN_ERROR);
+          } else if (path.parentPath.isMemberExpression()) {
+            const name = path.parent.property.name;
+
+            if (name === 'chain') {
+              throw path.buildCodeFrameError(CHAIN_ERROR);
+            }
+
+            if (!pkgStore.has(name)) {
+              const identifier = importModule(pkgStore, name, path.hub.file);
+              pkgStore.set(name, identifier);
+              pkgStore.set(identifier.name, identifier);
+            }
+          }
         });
+    });
+
+    // Phase 2: Replace all uses with their modularized version.
+    _.each(members, ({ pkgStore, spec }) => {
+      const name = spec.node.local.name;
+
+      const identifier = pkgStore.get(name);
+      const binding = spec.scope.getBinding(name);
+      _.each(binding.referencePaths, path => {
+        path.replaceWith(identifier);
+      });
+    })
+    _.each(defaults, ({ pkgStore, spec }) => {
+      const binding = spec.scope.getBinding(spec.node.local.name);
+      _.each(binding.referencePaths, path => {
+        const { parentPath, node } = path;
+
+        if (parentPath.isMemberExpression()) {
+          const name = path.parent.property.name;
+          parentPath.replaceWith(pkgStore.get(name));
+        } else {
+          const callee = getCallee(path);
+          if (callee && pkgStore.has(callee.name)) {
+            path.replaceWith(types.memberExpression(
+              callee,
+              types.identifier('placeholder')
+            ));
+          } else {
+            throw path.buildCodeFrameError('wtf');
+          }
```

Comment:
I managed to fix it (had to import everything before replacing any bindings), but I never updated my error message.